processor/processor.go

```diff
@@ -558,10 +559,14 @@ func (wp *workerPool) download(ctx context.Context, j *job.Job, validator *mimet
 		return derrors.E("creating request", err)
 	}
 
-	if j.UserAgent != "" {
-		req.Header.Set("User-Agent", j.UserAgent)
-	} else if wp.p.UserAgent != "" {
-		req.Header.Set("User-Agent", wp.p.UserAgent)
+	if len(j.RequestHeaders) == 0 {
+		for k, v := range wp.p.RequestHeaders {
+			req.Header.Set(k, v)
+		}
+	} else {
+		for k, v := range j.RequestHeaders {
+			req.Header.Set(k, v)
+		}
```

Comment:
I'd not expect a job that sets some headers, to cause the headers configured in the server to get discarded. In other words, I find it more reasonable to apply `wp.p.RequestHeaders` (if any) first and then merge any potential `j.RequestHeaders`.

Reply:
Done, added an explanatory comment also